gnucash/gnome-utils/dialog-preferences.c

```diff
@@ -189,16 +193,22 @@ gboolean
 gnc_account_separator_validate_cb (GtkEntry *entry, GdkEvent *event, GtkWidget *dialog)
 {
     gchar *separator;
-    gchar *conflict_msg = gnc_account_separator_is_valid (gtk_entry_get_text (entry), &separator);
+    gchar *conflict_msg;
```

Comment:
When you later test for the existence of this message, it's safer to explicitly initialize conflict_msg to NULL. Not all compilers to this by default so you may run into an uninitialized variable warning here (at best) or a NULL dereference at runtime (at worst).

Reply:
I agree a `=NULL` is safe and clean. All paths however have the value of `conflict_msg` set before use, so there is no possibility of NULL dereference and no compiler will warn here. However the variable `separator` is suspect as generally there is no promise compiler should know if the `gnc_account_separator_is_valid` will initialize it. It does, so there is no g_free of uninitialized pointer.

Reply:
Right. I got confused by reading the diff instead of the final code.

However in that case, my recommendation changes to: only define `conflict_msg` in the scope you use it, which s inside the `if (!invalid_separator_warned)` block.

Additionally, please use the same coding style as already in the file. That is put the `{` on a new line by itself.

Reply:
Ok. A few other functions like gnc_prefs_build_widget_table and gnc_prefs_find_page seemed to use style of declaring all variables at start of function, so I followed that. The brace is just a mistake of too much use like that.

Reply:
Heh, the old code is not always well-formed unfortunately. We're trying to do better these days, but there are still plenty of ugly bits around...